### PR TITLE
Add EnsureBackendService test

### PR DESCRIPTION
**Pull request by akwi-github**
**Setup**

1. Create  Global HealthCheck
2. Create a Global BackendService and add 1 HealthCheck to it

**Changes**

Execute planner only (without executing)

**Expectations**

Ensure that there are no necessary update actions

### CODE REVIEW

File: e2e/backend_service_update_test.go [hunk 1]
```diff
@@ -226,3 +226,39 @@ func TestBackendServiceUpdate(t *testing.T) {
...
+
+func TestEnsureBackendService(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	resUniqueId := "ensure-bs-test"
+	graphBuilder := rgraph.NewBuilder()
+	hcID, err := buildHealthCheck(graphBuilder, resUniqueId+"-hc", 15)
+	if err != nil {
+		t.Fatalf("buildHealthCheck(_, hc1-test, 15) = (_, %v), want (_, nil)", err)
```
Comment (kl52752) [inline e2e/backend_service_update_test.go:238]: `buildHealthCheck(_, %s-hc, 15), resUniqueId`

File: e2e/backend_service_update_test.go [hunk 2]
```diff
@@ -226,3 +226,39 @@ func TestBackendServiceUpdate(t *testing.T) {
 	wantBS = &compute.BackendService{LoadBalancingScheme: "INTERNAL_MANAGED"}
 	checkBackendService(t, ctx, theCloud, bs1ID, wantBS, compareLBScheme)
 }
+
+func TestEnsureBackendService(t *testing.T) {
```
Comment (kl52752) [inline e2e/backend_service_update_test.go:230]: ensure test should: 
1. create all resources 
2. call planing again on the same graph and verify that planning doesn't return action to perform (that means that GCE state is equal desired state)

Comment (akwi-github) [reply e2e/backend_service_update_test.go:230]: Please take another look.

File: e2e/backend_service_update_test.go (hunk 1 above)
Comment (akwi-github) [reply e2e/backend_service_update_test.go]: Done.

File: e2e/backend_service_update_test.go [hunk 3]
```diff
@@ -226,3 +226,63 @@ func TestBackendServiceUpdate(t *testing.T) {
...
+	bsID, err := buildBackendServiceWithLBScheme(graphBuilder, resUniqueId+"-bs", hcID, "INTERNAL_SELF_MANAGED")
+	if err != nil {
+		t.Fatalf("buildBackendServiceWithLBScheme(_, %s, _) = %v, want nil", resUniqueId+"-bs", err)
+	}
+	expectedActions := []exec.ActionMetadata{
+		{Type: exec.ActionTypeCreate, Name: actionName(exec.ActionTypeCreate, bsID)},
+		{Type: exec.ActionTypeCreate, Name: actionName(exec.ActionTypeCreate, hcID)},
+	}
+
+	graph, err := graphBuilder.Build()
```
Comment (kl52752) [inline e2e/backend_service_update_test.go:249]: check error

File: e2e/backend_service_update_test.go [hunk 4]
```diff
@@ -226,3 +226,63 @@ func TestBackendServiceUpdate(t *testing.T) {
...
+
+	processGraphAndExpectActions(t, graphBuilder, expectedActions)
+	checkGCEBackendService(t, ctx, theCloud, hcID, bsID, 80)
+
+	expectedActions = []exec.ActionMetadata{
+		{Type: exec.ActionTypeMeta, Name: eventName(bsID)},
+		{Type: exec.ActionTypeMeta, Name: eventName(hcID)},
+	}
+
+	graph, err = graphBuilder.Build()
```
Comment (kl52752) [inline e2e/backend_service_update_test.go:273]: check error

Comment (akwi-github) [reply e2e/backend_service_update_test.go:273]: Done.

File: e2e/backend_service_update_test.go (hunk 3 above)
Comment (akwi-github) [reply e2e/backend_service_update_test.go:249]: Done.

**Pull request merged**